### DISCUSSION TITLE
How to access props of children in the parent component ?

### DISCUSSION THREAD

**Discussion by iongion** [solidjs/solid#1765] [Q&A]
I understood that there is no vdom so not all things are possible like in similar frameworks, I am trying to achieve this

```typescript
<Tabs id="TabsExample">
  <Tab id="TabsExampleTabID1" title="Home">Home tab</Tab>
  <Tab id="TabsExampleTabID2" title="Files">Files tab</Tab>
  <Tab id="TabsExampleTabID3" title="Builds">Builds</Tab>
</Tabs>
```

- The `Tabs` component is able to internally traverse all children, filter those that are of type `Tab` and extract their `title` to actually build a `Tab navigation` bar

Visually, it gets represented like this

![image](https://github.com/solidjs/solid/assets/867609/6aae8df1-cca1-4a70-8aeb-782618af5c03)

So, what I am looking for is a way to access the `prop` `title` of a `Tab`, but from the parent `Tabs` component.

A way would be to not use that structure above, but I am porting an existing popular framework, I want that its documentation is kept, not surprising for the Solid developer adopting it.

**Comment by snnsnn** [comment]
You need to decide where to keep the state. If you are going to keep it inside `Tabs` then you need to pass necessary props and event handlers to children. You can pass them directly via props, or you can use Context which I strongly recommend against.

If you are going to keep the state inside `Tab` then you need to pass  the required information to the parent. You can use context api for that.

Since solid compiles to native DOM elements you can imperatively interact with the children but it is best to avoid because imperative API does not work on the server side, and it produces larger bundle size.

Solid provides a helper function called children, which might come handy if you are going to interact with reactive children: https://www.solidjs.com/docs/latest/api#children

So, options and interactions are similar to that of React's.
